sdks/java/extensions/smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSource.java

```diff
@@ -143,30 +136,39 @@ public SortedBucketSource(
     @SuppressWarnings("deprecation")
     final Reshuffle.ViaRandomKey<KV<Integer, List<BucketedInput<?, ?>>>> reshuffle =
         Reshuffle.viaRandomKey();
+
+    final List<TupleTag<?>> tupleTags =
+        sources.stream().map(i -> i.tupleTag).collect(Collectors.toList());
```

Comment:
we should probably add a getter function for `BucketedInput.tupleTag` so this stream syntax can match L147 (`sources.stream().map(BucketedInput::getCoder)`)